lib/features/family/features/reflect/presentation/pages/grateful_screen.dart

```diff
@@ -106,13 +106,16 @@ class _GratefulScreenState extends State<GratefulScreen> {
                   uiModel: uiModel.avatarBarUIModel,
                   onAvatarTapped: _cubit.onAvatarTapped,
                 ),
+                const SizedBox(height: 24),
                 Flexible(
                   child: RecommendationsWidget(
                     uiModel: uiModel.recommendationsUIModel,
                     onRecommendationChosen: (int i) {
-                      _cubit.onRecommendationChosen(i);
+                      _cubit.onRecommendationChosen(
+                          i, uiModel.recommendationsUIModel.showActsOfService);
```

Comment:
See other comment above. Also try to avoid passing the uiModel back into the cubit. Sometimes there's exceptions but the responsibility of the uimodel should only help the ui display itself visually.